controllers/asosecret_controller.go

```diff
@@ -279,22 +279,25 @@ func (asos *ASOSecretReconciler) createSecretFromClusterIdentity(ctx context.Con
 	newASOSecret.Data["AZURE_CLIENT_ID"] = []byte(identity.Spec.ClientID)
 
 	// Fetch identity secret, if it exists
-	key = types.NamespacedName{
-		Namespace: identity.Spec.ClientSecret.Namespace,
-		Name:      identity.Spec.ClientSecret.Name,
-	}
-	identitySecret := &corev1.Secret{}
-	err := asos.Get(ctx, key, identitySecret)
-	if err != nil && !apierrors.IsNotFound(err) {
-		return nil, errors.Wrap(err, "failed to fetch AzureClusterIdentity secret")
-	}
+	if identity.Spec.ClientSecret != (corev1.SecretReference{}) {
+		key = types.NamespacedName{
```

Comment:
The only change here is wrapping this block in the `if`, which the "hide whitespace" checkbox should make clear.

Reply:
Hmm I think `identity.Spec.ClientSecret` should actually be a pointer since it's marked as optional (`*corev1.SecretReference`) and then we'd be able to do `if identity.Spec.ClientSecret != nil`

Reply:
For now a more readable check might be `if  identity.Spec.ClientSecret.Name != ""` but IMO we should just change it to be pointer (and make sure there are no other places in the code that assume it's set)

Reply:
Is the pointer change something we would backport? Or would it make sense to include as a separate commit in this PR?

Reply:
Fair, if we don't want to backport the type change we can make it as a separate commit or even PR and only backport this one.

non-blocking, but I slightly prefer this for the cherry-pick, reads for easily as "not empty" to me, might be a personal preference though:

```suggestion
		if identity.Spec.ClientSecret.Name != "" {
```

Reply:
The fix part of this change probably doesn't make sense to cherry-pick since this entire controller only exists on the main branch. (added by #3698)

I'll make this PR 2 commits: one for the type change, then one for the fix.

Reply:
So no need to cherry-pick any of it then?

Reply:
I think not.